File: src/main/java/com/raejz/sc/bus/movie/model/Movie.java
package com.raejz.sc.bus.movie.model;

import com.google.auto.value.AutoValue;
import com.raejz.sc.bus.common.model.BaseDoc;
import java.util.List;

/**
 *
 */
@AutoValue
public class Movie extends BaseDoc {
  private Integer year;
  private String title;
  private String synopsis;
  private List<String> actorIds;
}

File: src/main/kotlin/com/raejz/sc/bus/movie/model/Movie.kt
package com.raejz.sc.bus.movie.model

import com.raejz.sc.bus.common.model.BaseDoc
import java.util.*

/**
 *
 */
data class Movie(
        val year: Int? = null,
        val title: String? = null,
        val synopsis: String? = null,
        val actorIds: List<String>? = null,
        override var createAt: Date? = null,
        override var modifyAt: Date? = null
) : BaseDoc()